### ISSUE TITLE
documentation outdated?

### ISSUE THREAD

**Issue by ldegen** [mclaeysb/simplepolygon#12]
Hi,

This tool is really helpful to me, thanks a lot!

I have a question regarding an edge case mentioned in the README.md:

>  An exception includes polygons with spikes or cuts such as [[0,0],[2,0],[1,1],[2,2],[0,2],[1,1],[0,0]], who 
> are currently allowed and treated correctly, but make the output non-simple (by OGC definition). 

This example is rejected in 1.1.7. Try to run this:

``` javascript
var feature, simplepolygon;

simplepolygon = require("simplepolygon");

feature = {
  type: "Feature",
  geometry: {
    type: "Polygon",
    coordinates: [[0, 0], [2, 0], [1, 1], [2, 2], [0, 2], [1, 1], [0, 0]]
  }
};

simplepolygon(feature);
```

Is this a bug? Or is the README outdated?

**Comment by mclaeysb** [comment]
Hi Idegen

Thanks for your interest in `simplepolygon`. I'm glad to hear it's of use to you - and I'm curious to know what your using it for.

Thanks for posting your exact input, it's a good practice and really helped me debugging. The main reason your getting the error is because your input is lacking an extra pair of square brackets around the coordinates. In GeoJSON, Polygon's are arrays of arrays points, in order to account for polygons with holes (see [here](https://macwright.org/2015/03/23/geojson-second-bite.html#polygons)). The examples I gave don't have this extra layer of []'s for simplicity, but that means I'm actually making it a little more complex - my apologies.

I've also noticed that the example I'm giving for a polygon with spike is not really a polygon with spike. I've changed to array to `[[0, 0], [2, 0], [0, 2], [4, 2], [2, 2], [0, 0]]` which [clearly](https://gist.github.com/anonymous/717eb9912d814856b53a25e160cc68d3) includes a spike.

These run fine in `simplepolygon`. There was a little but in the computation of the parents and the net winding number which I found by looking into your code, so I corrected it just now.

So `npm update` your directory and try out the following:

```
var feature, simplepolygon;

simplepolygon = require("simplepolygon");

feature = {
  type: "Feature",
  geometry: {
    type: "Polygon",
    coordinates: [[[0, 0], [2, 0], [0, 2], [4, 2], [2, 2], [0, 0]]]
  }
};

simplepolygon(feature);
```

Happy coding!

**Comment by ldegen** [comment]
Hi @mclaeysb, thanks for clarifying this.
I feel kind of stupid about the missing square brackets :-).

Looking at your new example, I also think I misread the statement from the README.md I was refering to.

So the algorithm does in fact reject *any* polygon that self-intersects in an existing vertex. 
It also rejects many other polygons that "are non-simple, by the OGC definition, for other reasons then self-intersection". It does however handle some of those edge cases gracefully, and your example with the spikes is such a case.

Correct?

>  I'm curious to know what your using it for.

You asked for it ;-)

I am experimenting with ways to create star-like shapes by rotating/mirroring simpler shapes.
It works a bit like a kaleidoscope. The input shapes are closed paths, created from "skinning" a tree-like
path that is in turn created from an l-system. The "branches" of that tree are expected to (self-)intersect, (which is actually desirable for aesthetic reasons). The "skinning"-algorithm is very simplistic and may create further self-intersections. Combining the mirrored/rotated copies will create more intersections.
I need the results to be "clean" because
  - I do post processing in other tools that require their input to be "simple"
  - I need to automate this step because it will be repeatet like several hundered times.
  - I need to detect cases where the generated shape contains disjoint "parts". 

I did try to build my own Bentley-Ottmann-based simplifier and failed miserably. :-)
`simplepolygon` handles about 75% of the cases, which is the best I got so far. I haven't tried Clipper yet.
Maybe I just find another way to work around the "bad" cases. This is basically a fun project, so there's no pressure.
